### PR TITLE
More type-hints and documentation added

### PR DESCRIPTION
**Pull request by Quix0r**
This PR contains more added type-hints, 1 hour straight no `TypeError` or anything related to these changes.

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -342,7 +343,7 @@ public static function markForDeletionById($item_id, $priority = PRIORITY_HIGH)
 
 		Post\Category::storeTextByURIId($item['uri-id'], $item['uid'], '');
 
-		if (!Post::exists(["`uri-id` = ? AND `uid` != 0 AND NOT `deleted`", $item['uri-id']])) {
+		if (!Post::exists(['`uri-id` = ? AND `uid` != 0 AND NOT `deleted`', $item['uri-id']])) {
```
Comment (MrPetovan) [inline src/Model/Item.php]: The convention is to use double-quote for SQL statements or conditions.
```suggestion
		if (!Post::exists(["`uri-id` = ? AND `uid` != 0 AND NOT `deleted`", $item['uri-id']])) {
```

File: src/Model/Item.php [hunk 2]
```diff
@@ -497,7 +511,7 @@ private static function isDuplicate(array $item)
 			}
 		} elseif ($item['network'] == Protocol::OSTATUS) {
 			// Check for an existing post with the same content. There seems to be a problem with OStatus.
-			$condition = ["`body` = ? AND `network` = ? AND `created` = ? AND `contact-id` = ? AND `uid` = ?",
+			$condition = ['`body` = ? AND `network` = ? AND `created` = ? AND `contact-id` = ? AND `uid` = ?',
```
Comment (MrPetovan) [inline src/Model/Item.php]:
```suggestion
			$condition = ["`body` = ? AND `network` = ? AND `created` = ? AND `contact-id` = ? AND `uid` = ?",
```

File: src/Model/Item.php [hunk 3]
```diff
@@ -623,15 +637,20 @@ public static function isTooOld(array $item)
...
+	 * @return integer Item id or zero on error
 	 */
-	private static function getDuplicateID(array $item)
+	private static function getDuplicateID(array $item): int
 	{
 		if (empty($item['network']) || in_array($item['network'], Protocol::FEDERATED)) {
-			$condition = ["`uri-id` = ? AND `uid` = ? AND `network` IN (?, ?, ?, ?)",
-				$item['uri-id'], $item['uid'],
-				Protocol::ACTIVITYPUB, Protocol::DIASPORA, Protocol::DFRN, Protocol::OSTATUS];
+			$condition = ['`uri-id` = ? AND `uid` = ? AND `network` IN (?, ?, ?, ?)',
```
Comment (MrPetovan) [inline src/Model/Item.php:646]:
```suggestion
			$condition = ["`uri-id` = ? AND `uid` = ? AND `network` IN (?, ?, ?, ?)",
```

File: src/Model/Nodeinfo.php [hunk 4]
```diff
@@ -156,12 +156,16 @@ public static function getServices()
...
+		$organization = [
+			'name' => null,
+			'contact' => null,
+			'account' => null
+		];
 
-		if (!empty($config->get('config', 'admin_email'))) {
-			$adminList = explode(',', str_replace(' ', '', $config->get('config', 'admin_email')));
+		if (!empty(DI::config()->get('config', 'admin_email'))) {
+			$adminList = explode(',', str_replace(' ', '', DI::config()->get('config', 'admin_email')));
```
Comment (MrPetovan) [inline src/Model/Nodeinfo.php]: Please keep the explicit dependency, `DI::config()` is a temporary crutch for the legacy static `Config` but it shouldn't replace an actual object.
```suggestion
		if (!empty($config()->get('config', 'admin_email'))) {
			$adminList = explode(',', str_replace(' ', '', $config()->get('config', 'admin_email')));
```

File: src/Model/Nodeinfo.php [hunk 5]
```diff
@@ -156,12 +156,16 @@ public static function getServices()
 		return $services;
 	}
 
-	public static function getOrganization($config)
+	public static function getOrganization()
```
Comment (MrPetovan) [inline src/Model/Nodeinfo.php]:
```suggestion
	public static function getOrganization(\Friendica\Core\Config\Capability\IManageConfigValues $config)
```

File: src/Model/OpenWebAuthToken.php [hunk 6]
```diff
@@ -82,10 +82,10 @@ public static function getMeta($type, $uid, $token)
 	 * @param string $interval SQL compatible time interval
 	 * @throws \Exception
 	 */
-	public static function purge($type, $interval)
+	public static function purge(string $type, string $interval)
 	{
-		$condition = ["`type` = ? AND `created` < ?", $type, DateTimeFormat::utcNow() . " - INTERVAL " . $interval];
-		DBA::delete("openwebauth-token", $condition);
+		$condition = ['`type` = ? AND `created` < ?', $type, DateTimeFormat::utcNow() . ' - INTERVAL ' . $interval];
```
Comment (MrPetovan) [inline src/Model/OpenWebAuthToken.php]:
```suggestion
		$condition = ["`type` = ? AND `created` < ?", $type, DateTimeFormat::utcNow() . ' - INTERVAL ' . $interval];
```

File: src/Model/Photo.php [hunk 7]
```diff
@@ -156,19 +156,19 @@ public static function getPhotoForUser($uid, $resourceid, $scale = 0, array $con
...
-		$uid = $r["uid"];
+		$uid = $r['uid'];
 
 		$accessible = $uid ? (bool)DI::pConfig()->get($uid, 'system', 'accessible-photos', false) : false;
 
 		$sql_acl = Security::getPermissionsSQLByUserId($uid, $accessible);
 
-		$conditions = ["`resource-id` = ? AND `scale` <= ? " . $sql_acl, $resourceid, $scale];
-		$params = ["order" => ["scale" => true]];
+		$conditions = ['`resource-id` = ? AND `scale` <= ? ' . $sql_acl, $resourceid, $scale];
```
Comment (MrPetovan) [inline src/Model/Photo.php]:
```suggestion
		$conditions = ["`resource-id` = ? AND `scale` <= ? " . $sql_acl, $resourceid, $scale];
```

File: src/Model/Photo.php [hunk 8]
```diff
@@ -331,71 +331,71 @@ public static function createPhotoForExternalResource($url, $uid = 0, $mimetype
 	 * @return boolean True on success
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	public static function store(Image $Image, $uid, $cid, $rid, $filename, $album, $scale, $type = self::DEFAULT, $allow_cid = "", $allow_gid = "", $deny_cid = "", $deny_gid = "", $desc = "")
+	public static function store(Image $image, int $uid, int $cid, string $rid, string $filename, string $album, int $scale, int $type = self::DEFAULT, string $allow_cid = '', string $allow_gid = '', string $deny_cid = '', string $deny_gid = '', string $desc = ''): bool
 	{
-		$photo = self::selectFirst(["guid"], ["`resource-id` = ? AND `guid` != ?", $rid, ""]);
+		$photo = self::selectFirst(['guid'], ['`resource-id` = ? AND `guid` != ?', $rid, '']);
```
Comment (MrPetovan) [inline src/Model/Photo.php]:
```suggestion
		$photo = self::selectFirst(['guid'], ["`resource-id` = ? AND `guid` != ?", $rid, '']);
```

File: src/Model/Photo.php [hunk 9]
```diff
@@ -331,71 +331,71 @@ public static function createPhotoForExternalResource($url, $uid = 0, $mimetype
 	 * @return boolean True on success
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	public static function store(Image $Image, $uid, $cid, $rid, $filename, $album, $scale, $type = self::DEFAULT, $allow_cid = "", $allow_gid = "", $deny_cid = "", $deny_gid = "", $desc = "")
+	public static function store(Image $image, int $uid, int $cid, string $rid, string $filename, string $album, int $scale, int $type = self::DEFAULT, string $allow_cid = '', string $allow_gid = '', string $deny_cid = '', string $deny_gid = '', string $desc = ''): bool
```
Comment (MrPetovan) [inline src/Model/Photo.php:334]: I prefer to capitalize variables holding structured objects as they are in minority and I feel like they should stand out.
```suggestion
	public static function store(Image $Image, int $uid, int $cid, string $rid, string $filename, string $album, int $scale, int $type = self::DEFAULT, string $allow_cid = '', string $allow_gid = '', string $deny_cid = '', string $deny_gid = '', string $desc = ''): bool
```

File: src/Model/Photo.php [hunk 10]
```diff
@@ -433,63 +433,63 @@ public static function delete(array $conditions, array $options = [])
...
+	 * @param Image $image      Image to update. Optional, default null.
+	 * @param array $old_fields Array with the old field values that are about to be replaced (true = update on duplicate)
 	 *
 	 * @return boolean  Was the update successfull?
 	 *
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 * @see   \Friendica\Database\DBA::update
 	 */
-	public static function update($fields, $conditions, Image $img = null, array $old_fields = [])
+	public static function update(array $fields, array $conditions, Image $image = null, array $old_fields = []): bool
```
Comment (MrPetovan) [inline src/Model/Photo.php:452]:
```suggestion
	public static function update(array $fields, array $conditions, Image $Image = null, array $old_fields = []): bool
```

File: src/Model/Photo.php [hunk 11]
```diff
@@ -507,66 +507,66 @@ public static function importProfilePhoto($image_url, $uid, $cid, $quit_on_error
...
+		if ($quit_on_error && ($img_str == '')) {
 			return false;
 		}
 
 		$type = Images::getMimeTypeByData($img_str, $image_url, $type);
 
-		$Image = new Image($img_str, $type);
-		if ($Image->isValid()) {
-			$Image->scaleToSquare(300);
+		$image = new Image($img_str, $type);
```
Comment (MrPetovan) [inline src/Model/Photo.php:516]:
```suggestion
		$Image = new Image($img_str, $type);
```

File: src/Model/Photo.php [hunk 12]
```diff
@@ -590,31 +590,33 @@ public static function importProfilePhoto($image_url, $uid, $cid, $quit_on_error
 	 * @param string $hemi      hemi
 	 * @return float
 	 */
-	public static function getGps($exifCoord, $hemi)
+	public static function getGps(array $exifCoord, strinf $hemi): float
```
Comment (MrPetovan) [inline src/Model/Photo.php]: Typo.
```suggestion
	public static function getGps(array $exifCoord, string $hemi): float
```

File: src/Model/Photo.php [hunk 13]
```diff
@@ -885,17 +889,23 @@ public static function isLocalPage($name)
...
 	}
 
-	private static function fitImageSize($Image)
+	/**
+	 * Tries to resize image to wanted maximum size
+	 *
+	 * @param Image $image Image instance
+	 * @return Image|null Image instance on success, null on error
+	 */
+	private static function fitImageSize(Image $image)
```
Comment (MrPetovan) [inline src/Model/Photo.php:898]:
```suggestion
	private static function fitImageSize(Image $Image)
```

File: src/Model/Photo.php [hunk 14]
```diff
@@ -939,17 +955,23 @@ private static function loadImageFromURL(string $image_url)
 
 		$type = Images::getMimeTypeByData($img_str, $image_url, $type);
 
-		$Image = new Image($img_str, $type);
+		$image = new Image($img_str, $type);
```
Comment (MrPetovan) [inline src/Model/Photo.php:958]:
```suggestion
		$Image = new Image($img_str, $type);
```

File: src/Model/Photo.php [hunk 15]
```diff
@@ -1008,34 +1030,36 @@ private static function uploadImage(array $files)
 		Logger::info('File upload', ['src' => $src, 'filename' => $filename, 'size' => $filesize, 'type' => $filetype]);
 
 		$imagedata = @file_get_contents($src);
-		$Image = new Image($imagedata, $filetype);
-		if (!$Image->isValid()) {
+		$image = new Image($imagedata, $filetype);
```
Comment (MrPetovan) [inline src/Model/Photo.php:1033]:
```suggestion
		$Image = new Image($imagedata, $filetype);
```

File: src/Model/Photo.php [hunk 16]
```diff
@@ -1052,10 +1076,10 @@ public static function upload(int $uid, array $files, string $album = '', string
 			return [];
 		}
 
-		$Image    = $data['image'];
+		$image    = $data['image'];
```
Comment (MrPetovan) [inline src/Model/Photo.php:1079]:
```suggestion
		$Image    = $data['image'];
```

File: src/Model/Photo.php [hunk 17]
```diff
@@ -1139,10 +1163,10 @@ public static function uploadAvatar(int $uid, array $files, string $url = ''): s
 			return '';
 		}
 
-		$Image    = $data['image'];
+		$image    = $data['image'];
```
Comment (MrPetovan) [inline src/Model/Photo.php:1166]:
```suggestion
		$Image    = $data['image'];
```

File: src/Model/Photo.php [hunk 18]
```diff
@@ -1217,19 +1241,19 @@ public static function uploadBanner(int $uid, array $files = [], string $url = '
...
```
Comment (MrPetovan) [inline src/Model/Photo.php:1244]:
```suggestion
		$Image    = $data['image'];
```

File: src/Module/NodeInfo210.php [hunk 19]
```diff
@@ -57,7 +57,7 @@ protected function rawContent(array $request = [])
 				'software' => 'friendica',
 				'version'  => FRIENDICA_VERSION . '-' . DB_UPDATE_VERSION,
 			],
-			'organization'      => Nodeinfo::getOrganization($this->config),
+			'organization'      => Nodeinfo::getOrganization(),
```
Comment (MrPetovan) [inline src/Module/NodeInfo210.php]:
```suggestion
			'organization'      => Nodeinfo::getOrganization($this->config),
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Model/Item.php (hunk 2 above)
Comment (Quix0r) [reply src/Model/Item.php]: Do you want me to fix the indenting here?

File: src/Model/Item.php (hunk 3 above)
Comment (Quix0r) [reply src/Model/Item.php:646]: You have `$condition = ['`` on more places:
- PostUpdate.php, lines 140, 347, 1441
- OpenWebAuthToken.php, line 87
- Item.php, line 646
- Contact.php, lines 274, 995
- DRN.php, lines 230, 2324

Do you want me to fix them all?

File: src/Model/Nodeinfo.php (hunk 5 above)
Comment (Quix0r) [reply src/Model/Nodeinfo.php]: I usually like to have these long class names (with namespace) in top of the file with a proper `use` statement and then have the source code smaller.

File: src/Model/Photo.php (hunk 9 above)
Comment (Quix0r) [reply src/Model/Photo.php:334]: No, let's make all variables the same, please no extra rules. :-)

File: src/Model/Nodeinfo.php (hunk 5 above)
Comment (MrPetovan) [reply src/Model/Nodeinfo.php]: Of course, but I can't make fancy code suggestions like these. Feel free to add the namespace in a `use` statement.

File: src/Model/Photo.php (hunk 9 above)
Comment (MrPetovan) [reply src/Model/Photo.php:334]: How do you feel about it, @annando  and @nupplaphil ?

File: src/Model/Nodeinfo.php (hunk 5 above)
Comment (Quix0r) [reply src/Model/Nodeinfo.php]: Fixed this + missing doctag and return type.

File: src/Model/Item.php (hunk 2 above)
Comment (MrPetovan) [reply src/Model/Item.php]: No, it's fine, the values are following the SQL snippet order, it's better this way.

**Review by MrPetovan — APPROVED** [review]

File: src/Model/Item.php (hunk 3 above)
Comment (MrPetovan) [reply src/Model/Item.php:646]: No thank you, I'd rather you only focus on the files you've already edited.

**Comment by MrPetovan** [comment]
From Woodpecker:
> ```
> 1) Friendica\Test\src\Module\Api\Friendica\NotificationTest::testWithXmlResult
> Failed asserting that two DOM documents are equal.
> --- Expected
> +++ Actual
> @@ @@
> 
> -
> +
> 
> /woodpecker/src/github.com/friendica/friendica/tests/src/Module/Api/Friendica/NotificationTest.php:72
> ```

**Comment by Quix0r** [comment]
Same problem here. I don't know what have caused this.

**Comment by Quix0r** [comment]
@MrPetovan @nupplaphil @annando Do you know how to fix that error coming from the unit test?

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**